pkg/tsdb/azuremonitor/types/types.go

```diff
@@ -109,21 +109,26 @@ type AzureResponseTable struct {
 // AzureMonitorJSONQuery is the frontend JSON query model for an Azure Monitor query.
 type AzureMonitorJSONQuery struct {
 	AzureMonitor struct {
-		Aggregation         string  `json:"aggregation"`
-		Alias               string  `json:"alias"`
+		ResourceURI     string `json:"resource"`
+		MetricNamespace string `json:"metricNamespace"`
+		MetricName      string `json:"metricName"`
+
+		Aggregation      string                        `json:"aggregation"`
+		Alias            string                        `json:"alias"`
+		DimensionFilters []AzureMonitorDimensionFilter `json:"dimensionFilters"` // new model
+		TimeGrain        string                        `json:"timeGrain"`
+		Top              string                        `json:"top"`
+
+		MetricDefinition string `json:"metricDefinition"`
+		ResourceGroup    string `json:"resourceGroup"`
+		ResourceName     string `json:"resourceName"`
+
+		// Legecy "resource" fields from before the resource picker provided just a single ResourceURI
+		// These are used for pre-resource picker queries to reconstruct a resource URI
```

Comment:
I'm not super clear on this comment, I know its carried over from the draft but I'm not sure it's accurate? 

I would think we would still pass along time grains and format for example even when we have the resource uri right like the uri doesn't reallly capture things like that? I'm not sure what dimension and dimension filter would be in this context

Reply:
I think it's actually meant for the group of params above it.